examples/oven-app/silabs/src/OvenUI.cpp

```diff
@@ -0,0 +1,150 @@
+{
+    // Draw Silabs Corner icon
+    GLIB_drawBitmap(glibContext, SILABS_ICON_POSITION_X, STATUS_ICON_LINE, SILABS_LOGO_WIDTH, SILABS_LOGO_HEIGHT, silabsLogo);
+    // Draw BLE Icon
+    GLIB_drawBitmap(glibContext, BLE_ICON_POSITION_X, STATUS_ICON_LINE, BLUETOOTH_ICON_SIZE, BLUETOOTH_ICON_SIZE, bleLogo);
+    // Draw WiFi/OpenThread Icon
+#ifdef SL_WIFI
+    GLIB_drawBitmap(glibContext, NETWORK_ICON_POSITION_X, STATUS_ICON_LINE, WIFI_BITMAP_WIDTH, WIFI_BITMAP_HEIGHT, wifiLogo);
+#else
+    GLIB_drawBitmap(glibContext, NETWORK_ICON_POSITION_X, STATUS_ICON_LINE, THREAD_BITMAP_WIDTH, THREAD_BITMAP_HEIGHT, threadLogo);
+#endif // SL_WIFI
+    // Draw Matter Icon
+    GLIB_drawBitmap(glibContext, MATTER_ICON_POSITION_X, STATUS_ICON_LINE, MATTER_LOGO_WIDTH, MATTER_LOGO_HEIGHT, matterLogoBitmap);
+}
+
+void OvenUI::DrawCookTopState(GLIB_Context_t * glibContext)
+{
+    // Get CookTop state from OvenManager
+    bool cookTopState = OvenManager::GetInstance().GetCookTopState();
+
+    if (cookTopState)
+    {
+        GLIB_drawStringOnLine(glibContext, "COOKTOP: ON", COOKTOP_STATE_DISPLAY_LINE, GLIB_ALIGN_LEFT, 0, 0, true);
+    }
+    else
+    {
+        GLIB_drawStringOnLine(glibContext, "COOKTOP: OFF", COOKTOP_STATE_DISPLAY_LINE, GLIB_ALIGN_LEFT, 0, 0, true);
+    }
+}
+
+void OvenUI::DrawOvenMode(GLIB_Context_t * glibContext)
+{
+    // Get current oven mode from the Temperature Controlled Cabinet endpoint
+    uint8_t currentMode = OvenManager::GetInstance().GetCurrentOvenMode();
+
+    // LCD Display line max length is 16 characters, so allocate 17 bytes for the message (MODE: + mode text + null terminator)
+    char message[17];
+    const char * modeText;
+
+    switch (currentMode)
+    {
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeBake):
+        modeText = "BAKE";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeConvection):
+        modeText = "CONVECTION";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeGrill):
+        modeText = "GRILL";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeRoast):
+        modeText = "ROAST";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeClean):
+        modeText = "CLEAN";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeConvectionBake):
+        modeText = "CONV BAKE";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeConvectionRoast):
+        modeText = "CONV ROAST";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeWarming):
+        modeText = "WARMING";
+        break;
+    case chip::to_underlying(OvenModeDelegate::OvenModes::kModeProofing):
+        modeText = "PROOFING";
+        break;
+    default:
+        modeText = "UNKNOWN";
+        break;
+    }
+
+    snprintf(message, sizeof(message), "MODE: %s", modeText);
```

Comment:
This is unsafe as you're removing the null character in some string which are bigger than the buffer size. this can lead to unwanted behavior down the line since the rest of the code expect a valid string....